component-runtime-manager/src/main/java/org/talend/sdk/component/runtime/manager/service/LocalCacheService.java

```diff
@@ -17,71 +17,236 @@
+        );
+    }
+
+    @Override
+    public T computeIfAbsent(String key, Supplier<T> value) {
+        final CacheConfiguration config = this.configuration.get();
+        long timeOut = config.isActive() ? config.getDefaultEvictionInterval() : -1;
+        return computeIfAbsent(key, timeOut, value);
     }
 
     private String internalKey(final String key) {
         return plugin + '@' + key;
     }
 
-    @AllArgsConstructor
-    public static class Element {
+    public void clean() {
+        final List<String> removableElements = this.cache
+                .entrySet()
+                .stream()
+                .filter(e -> e.getValue().mustBeRemoved())
+                .map(Entry::getKey)
+                .collect(toList());// materialize before actually removing it
+        removableElements.forEach(this.cache::remove);
+    }
+
+    public boolean isEmpty() {
+        return this.cache.isEmpty();
+    }
+
+    private ScheduledExecutorService getThreadService() {
+        if (this.threadService == null) {
+            synchronized (this) {
+                if (this.threadService == null) {
+                    this.threadService = Executors.newSingleThreadScheduledExecutor((Runnable r) -> {
+                        final Thread thread = new Thread(r, this.getClass().getName() + "-eviction-" + hashCode());
+                        thread.setPriority(Thread.NORM_PRIORITY);
+                        return thread;
+                    });
+                }
+            }
+        }
+        return this.threadService;
+    }
+
+    private ScheduledFuture<?> evictionTask(String key, long end) {
+        ScheduledFuture<?> task = null;
+        final CacheConfiguration config = this.configuration.get();
+        if (config != null && config.isActive()) {
+            task = this.getThreadService().schedule(() -> this.evict(key), end, SECONDS);
+        }
+        return task;
+    }
+
+    /**
+     * Cache configuration.
+     */
+    @Data
+    public static class CacheConfiguration implements Serializable {
+
+        @Option
+        private long defaultEvictionInterval;
+
+        @Option
+        private boolean active;
+    }
+
+    /**
+     * Wrapper for each cached object.
+     * @param <T> : type of cached object.
+     */
+    private static class ElementImpl<T> implements Element<T> {
 
-        private final Object value;
+        /** cached object */
+        private final T value;
 
+        /** function, if exists, that authorize to remove object. */
+        private final Predicate<Element<T>> toRemove;
+
+        /** give time object can be release (infinity if < 0) */
         private final long endOfValidity;
 
-        private boolean isExpired() {
-            return System.currentTimeMillis() > endOfValidity;
+        /** scheduled task to remove object if nedeed (to cancel if removed before) */
+        private final ScheduledFuture<?> removedTask;
+
+        public ElementImpl(Supplier<T> value,
```

Comment:
finals ;)

Reply:
done, even if it's useless on method parameter.